src/widgets/DataTable/DataTableRow.js

```diff
@@ -42,187 +42,197 @@ import { formatStatus } from '../../utilities/index';
-        editableCellText,
-        editableCellTextView,
-        editableCellUnfocused,
-      } = newDataTableStyles;
+    const rowStyle = rowIndex % 2 === 0 ? alternateRow : row;
 
-      // Map each column to an appropriate cell for a given row.
-      return columns.map(({ key: columnKey, type, width, alignText }, index) => {
-        const isLastCell = index === columns.length - 1;
-        const isDisabled = isFinalised || (rowState && rowState.isDisabled) || rowData.isFinalised;
-        const isFocused = focusedColumnKey === columnKey;
-        const cellAlignment = alignText || 'left';
-        switch (type) {
-          case 'editable':
-            return (
-              <TextInputCell
-                key={columnKey}
-                value={rowData[columnKey]}
-                rowKey={rowKey}
-                columnKey={columnKey}
-                editAction={getAction(columnKey)}
-                isFocused={isFocused}
-                isDisabled={isDisabled}
-                dispatch={dispatch}
-                width={width}
-                viewStyle={cellContainer[cellAlignment]}
-                textViewStyle={editableCellTextView}
-                isLastCell={isLastCell}
-                keyboardType="numeric"
-                textInputStyle={cellText[cellAlignment]}
-                textStyle={editableCellUnfocused[cellAlignment]}
-                cellTextStyle={editableCellText}
-                rowIndex={rowIndex}
-              />
-            );
+    // Callback for rendering a row of cells.
+    const renderCells = useCallback(
+      () =>
+        // Map each column to an appropriate cell for a given row.
+        columns.map(({ key: columnKey, type, width, alignText }, index) => {
+          // Indicator if the right hand border should be removed from styles for this cell.
+          const isLastCell = index === columns.length - 1;
 
-          case 'checkable':
-            return (
-              <CheckableCell
-                key={columnKey}
-                rowKey={rowKey}
-                columnKey={columnKey}
-                isChecked={rowState && rowState.isSelected}
-                isDisabled={isDisabled}
-                CheckedComponent={CheckedComponent}
-                UncheckedComponent={UncheckedComponent}
-                DisabledCheckedComponent={DisabledCheckedComponent}
-                DisabledUncheckedComponent={DisabledUncheckedComponent}
-                onCheckAction={getAction(columnKey, 'onCheckAction')}
-                onUncheckAction={getAction(columnKey, 'onUncheckAction')}
-                dispatch={dispatch}
-                containerStyle={touchableCellContainer}
-                width={width}
-                isLastCell={isLastCell}
-              />
-            );
+          // This cell is disabled if the pageObject is finalised, the row has been explicitly set
+          // as disabled, or the rowData is disabled (i.e. data is an invoice),
+          const isDisabled =
```

Comment:
Coudl destructure here just for consistency:

```
const { isDisabled: isRowStateDisabled } = rowState;
const { isDisabled: isRowDataDisabled } = rowData;
const isDisabled = isFinalised || isRowStateDisabled || isRowDataDisabled
```

Not sure if should be `rowData.isDisabled` or `rowData.isFinalised`? Looks like comment says the former, but existing code says the latter?
```

Reply:
comment wrong